src/Command/BaseGenerateCommand.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Yiisoft\Yii\Gii\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Yiisoft\Yii\Console\ExitCode;
+use Yiisoft\Yii\Gii\GeneratorInterface;
+use Yiisoft\Yii\Gii\GiiInterface;
+
+abstract class BaseGenerateCommand extends Command
+{
+    protected const NAME = '';
+    protected GiiInterface $gii;
+
+    public function __construct(GiiInterface $gii)
+    {
+        parent::__construct();
+        $this->gii = $gii;
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument('overwrite', InputArgument::OPTIONAL, '', false);
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|void|null
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $generator = $this->gii->getGenerator(static::NAME);
+        $generator->controllerClass = $input->getArgument('className');
+        echo "Running '{$generator->getName()}'...\n\n";
```

Comment:
Please use `$output` in all similar places

Reply:
yeah